src/GpuLidarSensor.cc

```diff
@@ -360,73 +360,58 @@ void GpuLidarSensorPrivate::FillPointCloudMsg(const float *_laserBuffer)
   IGN_PROFILE("GpuLidarSensorPrivate::FillPointCloudMsg");
   uint32_t width = this->pointMsg.width();
   uint32_t height = this->pointMsg.height();
-  unsigned int channels = 3;
-
-  float angleStep =
-    (this->gpuRays->AngleMax() - this->gpuRays->AngleMin()).Radian() /
-    (this->gpuRays->RangeCount()-1);
-
-  float verticleAngleStep = (this->gpuRays->VerticalAngleMax() -
-      this->gpuRays->VerticalAngleMin()).Radian() /
-    (this->gpuRays->VerticalRangeCount()-1);
-
-  // Angles of ray currently processing, azimuth is horizontal, inclination
-  // is vertical
-  float inclination = this->gpuRays->VerticalAngleMin().Radian();
+  unsigned int channels = this->gpuRays->Channels();
 
   std::string *msgBuffer = this->pointMsg.mutable_data();
-  msgBuffer->resize(this->pointMsg.row_step() *
-      this->pointMsg.height());
+  msgBuffer->resize(this->pointMsg.row_step() * height);
   char *msgBufferIndex = msgBuffer->data();
   // Set Pointcloud as dense. Change if invalid points are found.
-  bool isDense { true };
+  bool isDense = true;
+
   // Iterate over scan and populate point cloud
   for (uint32_t j = 0; j < height; ++j)
   {
-    float azimuth = this->gpuRays->AngleMin().Radian();
-
     for (uint32_t i = 0; i < width; ++i)
     {
       // Index of current point, and the depth value at that point
       auto index = j * width * channels + i * channels;
-      float depth = _laserBuffer[index];
-      // Validate Depth/Radius and update pointcloud density flag
-      if (isDense)
-        isDense = !(ignition::math::isnan(depth) || std::isinf(depth));
 
+      float depth = _laserBuffer[index + 0];
```

Comment:
Sorry I am not very clear about the code. How is it different from the one in `void Ogre2GpuRays::UpdateSampleTexture()` in `gz-rendering`?

Reply:
I mean the purpose of the code.

Reply:
Instead of recalculating azimuth and zenith for the samples, we grab it from the sample itself (because we had already calculated it when we generated it). Technically we could do the azimuth/zenith/depth to x/y/z conversion there as well (and directly pass x/y/z), but this way we keep changes to a minimum. Either way is fine though.